src/internal/base-component/telemetry.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getVisualTheme } from '../utils/get-visual-theme';
+import { useRuntimeVisualRefresh } from '../visual-mode';
+import { useComponentMetrics } from './component-metrics';
+import { ComponentConfiguration } from './metrics/interfaces';
+
+import { Environment } from './interfaces';
+
+export function createUseTelemetry(environment: Environment) {
+  return (componentName: string, config?: ComponentConfiguration) => {
+    const isVisualRefresh = environment.alwaysVisualRefresh || useRuntimeVisualRefresh();
```

Comment:
```
    const isVisualRefresh = environment.alwaysVisualRefresh ?? useRuntimeVisualRefresh();
```

because when `alwaysVisualRefresh` is `false` you do not want to fallback to runtime

Reply:
In classic this value would be false in the environment file, why wouldn't we want to fallback to runtime?

Reply:
This is why: https://unpkg.com/browse/@awsui/components-react@3.0.1475/internal/environment.js